src/utils/formula.js

```diff
@@ -144,7 +144,7 @@ const dBench = (dur_min, dur_marg, W) => {
 const userScore = (u1, y, userDoc) => {
   let user_score = u1 * y ** USER_SCORE_WEIGHT.W_Y;
   if (userDoc.blocked_by_admin?.status) {
-    user_score *= USER_SCORE_WEIGHT.W_PU;
+    user_score *= USER_SCORE_WEIGHT.W_PU * (userDoc.w_pu_user || 1);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for calculating `user_score` seems to have changed. Previously, if a user was blocked by an admin, the `user_score` was multiplied by `USER_SCORE_WEIGHT.W_PU`. Now, it's also being multiplied by `userDoc.w_pu_user || 1`. This means that if `userDoc.w_pu_user` is not defined or falsy (e.g., 0), it will default to 1 and won't affect the score. However, if `userDoc.w_pu_user` is defined and truthy, it could potentially change the user's score. 

If this is the intended behavior, then the code looks good. But if `userDoc.w_pu_user` should only affect the score when the user is not blocked by an admin, then the condition needs to be adjusted.

Here's how you can adjust the condition:

```diff
- if (userDoc.blocked_by_admin?.status) {
-     user_score *= USER_SCORE_WEIGHT.W_PU * (userDoc.w_pu_user || 1);
- }
+ if (userDoc.blocked_by_admin?.status) {
+     user_score *= USER_SCORE_WEIGHT.W_PU;
+ } else {
+     user_score *= (userDoc.w_pu_user || 1);
+ }
```

This way, `userDoc.w_pu_user` will only affect the score when the user is not blocked by an admin.


<!-- This is an auto-generated comment by OSS CodeRabbit -->